processor/schemaprocessor/internal/schema/manager.go

```diff
@@ -0,0 +1,209 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"sync"
+
+	schemaencoder "go.opentelemetry.io/otel/schema/v1.0"
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+const (
+	DefaultConcurrentFetches = 10
+)
+
+var (
+	errNilValueProvided = errors.New("nil value provided")
+)
+
+// Manager is responsible for ensuring that schemas are kept up to date
+// with the most recent version that are requested.
+type Manager interface {
+	// RequestSchema will returns a schema that can be used to
+	// transform attribute values or resource names.
+	// In the event that the schemaURL does not exist locally
+	// it will enqueued to be fetch and updated.
+	// Operations on the returned schema will be blocked
+	// until the update is complete
+	RequestSchema(schemaURL string) Schema
+
+	// ProcessRequest will perform the network operation
+	// of fetching the schema definitions
+	// and updating the schema being used.
+	ProcessRequests(ctx context.Context) error
```

Comment:
Is this a long running function?

Reply:
Yeah, the intention that is operates in the background handling any requests to fetch new information